assets/global.js

```diff
@@ -755,6 +755,41 @@ class VariantSelects extends HTMLElement {
     this.addEventListener('change', this.onVariantChange);
   }
 
+  updateVariantStatuses() {
+    const selectedOptionOneSelector = this.id.includes('variant-radios') ? '[type=radio]:checked+label' : 'option:checked';
+    const typeOfInput = this.id.includes('variant-radios') ? 'input' : 'option';
+
+    //Grab all the existing variants based on the selected option 1
+    this.selectedOptionOneVariants = JSON.parse(this.querySelector('[type="application/json"]').textContent).filter(variant => this.querySelector(selectedOptionOneSelector).innerText.trim() === variant.option1);
+    //Array of all the input wrappers we have (option1, option2 and option3)
+    this.inputWrappers = [...this.querySelectorAll('.product-form__input')];
+    //If there is only option 1 inputs then early return, don't check the rest
+    if (!this.inputWrappers[1]) return;
+    //Select the inputs present in the HTML
+    this.option2Inputs = [...this.inputWrappers[1].querySelectorAll(typeOfInput)];
+    //Create an array of all the available option 2 values of the currently selected option 1
+    this.selectedExistingOptions2 = this.selectedOptionOneVariants.filter(option => option.available).map( a => a.option2);
+    //For each input in the HTML check that the value is available in the array of available variants. If not, then add a class of disabled
+    this.option2Inputs.forEach(input => {
+      if (this.selectedExistingOptions2.includes(input.getAttribute('value'))) {
+        this.id.includes('variant-radios') ? input.classList.remove('disabled') : input.innerText = input.getAttribute('value');
+      } else {
+        this.id.includes('variant-radios') ? input.classList.add('disabled') : input.innerText = input.getAttribute('value') + ` - ${window.variantStrings.unavailable}`;
```

Comment:
Here I'm hard coding the `-` followed by the translation string for unavailable. But I should create a new string for `window.variantStrings` that points to `"value_unavailable": "{{ option_value }} - Unavailable"`since if a merchant change their translation in the admin, to a different format it won't update the JS.

Reply:
Changes for this are done 👍